acceptanceccl: fix restore tests

benesch opened the pull request:
See individual commit messages for details.

benesch committed:
acceptance/terrafarm: fix typo

benesch committed:
acceptanceccl: use sane VMs to run ccl tests

Azure D-series VMs have insane throttling that sends write latency off a
cliff as soon as the write throughput to the temporary disk exceeds
96MB/s. This causes cascading node liveness failures that grind restore
to halt. It turns out that `kv.bulk_io_write.max_rate` isn't as useful
as one would hope, as you need to set it low enough (~10MB/s) that the
combined throughput of bulk I/O, compactions, and flushes never exceeds
96MB/s. This makes these tests far slower than the need to be.

I'm working on using the RocksDB `rate_limiter` option so we can apply a
node-global (or store-global) write throughput limit. In the meantime,
though, just use Ls-series VMs, which aren't subject to the VM-level
throttling.

benesch committed:
acceptanceccl: remove stale cockroachDiskSizeGB option

This option has been dead since 9fc30f5 moved the data directory to
local SSD and the associated terraform variable. 8bec3b1 cleaned up most
references to [Cc]ockroachDiskSizeGB, but a few in acceptanceccl were
missed. Remove them.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17624)
<!-- Reviewable:end -->

dt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r2, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-KrONTHx-0MaP1rn0bx5:bnfp4nl)*
<!-- Sent from Reviewable.io -->

benesch committed:
acceptanceccl: colocate vms with storage accounts

Avoid egress bandwidth charges by colocating VMs for acceptanceccl tests
with the storage accounts they reference.

tamird commented:
Reviewed 1 of 1 files at r1, 3 of 3 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/acceptance/terrafarm/run.go, line 62 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-KrX6CAHVTVoERERZwlM:-KrX6CAHVTVoERERZwlN:bzb72pe) ([raw file](https://github.com/cockroachdb/cockroach/blob/e057cd06b6a0ac45a5f2148ab9af85e312cbb5e3/pkg/acceptance/terrafarm/run.go#L62)):*
> ```Go
> 		"-state="+f.StateFile,
> 		`-var=prefix="`+f.Prefix+`"`)
> 	return append(args, f.TerraformArgs...)
> ```

note that this will have a merge conflict with #17457.

---

*[pkg/ccl/acceptanceccl/backup_test.go, line 281 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-KrX6S7lYuoE5HSvcy7_:-KrX6S7lYuoE5HSvcy7a:b-66u57o) ([raw file](https://github.com/cockroachdb/cockroach/blob/ef56c3961112560eca466219b9a50de0bcff9600/pkg/ccl/acceptanceccl/backup_test.go#L281)):*
> ```Go
> 	bt := benchmarkTest{
> 		b:                   b,
> 		nodes:               10,
> ```

consider adding a comment explaining this value, particularly if you predict that a different VM class may be used in the future.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624)*
<!-- Sent from Reviewable.io -->

benesch committed:
acceptanceccl: disable stats-based rebalancing during tests

Stats-based rebalancing interacts badly with restore at the moment, so
disable it for now. Tracked in #17671.

tamird commented:
Reviewed 3 of 3 files at r6, 2 of 2 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624)*
<!-- Sent from Reviewable.io -->

benesch committed:
build: expose all acceptanceccl tests in teamcity-nightly-acceptance.sh

benesch commented:
Ok, PTAL! Had to add a few more assorted fixes.

Here's manually-triggered runs of RestoreTPCH10 and Restore2TB:

  * https://teamcity.cockroachdb.com/viewLog.html?buildId=330550&buildTypeId=Cockroach_Nightlies_RestoreTpch10&tab=buildLog
  * https://teamcity.cockroachdb.com/viewLog.html?buildId=330552&buildTypeId=Cockroach_Nightlies_Restore2TB&tab=buildLog

/cc @danhhz)

benesch commented:
Hmm, looks like 2TB causes a liveness failure after ~5% of progress, but the test seems correct, at least.

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r12, 5 of 5 files at r18, 2 of 2 files at r19, 4 of 4 files at r20, 1 of 1 files at r21.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-Ks9ghXGYEaBcQCRjqqK:bnfp4nl)*
<!-- Sent from Reviewable.io -->

danhhz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17624#-:-Ks9qnpB6QlqImEo0kTh:bnfp4nl)*
<!-- Sent from Reviewable.io -->

benesch commented:
TFTRs!

a-robinson commented on pkg/ccl/acceptanceccl/backup_test.go:
```diff
@@ -127,6 +143,13 @@ func (bt *benchmarkTest) Start(ctx context.Context) {
+	// TODO(benesch): Remove these settings when #17671 is fixed, or document the
+	// necessity of these settings as 1.1 known limitation.
+	sqlDB.Exec(`SET CLUSTER SETTING kv.allocator.stat_based_rebalancing.enabled = false`)
+	sqlDB.Exec(`SET CLUSTER SETTING kv.allocator.load_based_lease_rebalancing.enabled = false`)
```
@benesch I understand stat-based rebalancing caused problems, but was this setting independently shown to be an issue?
